### DISCUSSION TITLE
Question: Is zephyrproject actively maintaining the windows-curses sub-project?

### DISCUSSION THREAD

**Discussion by pjfarleyiii** [zephyrproject-rtos/zephyr#35409] [Q&A]
There have been several issues filed in the zephyrproject-rtos/windows-curses sub-project since June 2020 with no responses.  I am asking here whether zephyrproject-rtos intends to actively maintain the windows-curses sub-project or whether issues in windows-curses must be resolved via fork and independent debugging and corrections.

Peter

**Comment by carlescufi** [comment]
@pjfarleyiii thanks for the interest. The windows curses package was originally developed to enable menuconfig on Windows, and a new release has been made with the latest (3.9) Python repository. While the original maintainer is no longer involved with Zephyr, I do believe that maintaining the respository's location as the reference one is within the intent of the project. So I would encourage anyone to send Pull Requests and try to help with the issues, and we will make releases of it as needed.

**Reply by carlescufi** [reply]
We just updated windows-curses to version 2.3.0 with a more recent version of PDCurses and support for Python 3.10.

**Comment by pjfarleyiii** [comment]
I guess my question then becomes:  What can a "just plain python user" do when an issue s/he has found in windows-curses is blocking their own work, and they do not have the skills or experience to tackle debugging code as deep and complex as curses seems to be (especially the underlying PDCurses library)?

I would personally be more than happy to make a generous donation to anyone who would do that deep debugging work for windows-curses users and make windows-curses better for everyone who needs it.

Regards,

Peter

**Reply by carlescufi** [reply]
>I guess my question then becomes: What can a "just plain python user" do when an issue s/he has found in windows-curses is blocking their own work, and they do not have the skills or experience to tackle debugging code as deep and complex as curses seems to be (especially the underlying PDCurses library)?

My suggestion @pjfarleyiii: Look around for people in the PDCurses repo that might be able to debug your issues, and offer them contract work for that.

**Reply by carlescufi** [reply]
Also give the new 2.3.0 version a try, we updated PDCurses.

**Reply by pjfarleyiii** [reply]
> Also give the new 2.3.0 version a try, we updated PDCurses.

Thanks for the update, I will do that and report back.
